pkg/source/adapter/adapter.go

```diff
@@ -303,3 +307,14 @@ func (a *Adapter) InitOffsets(session sarama.ConsumerGroupSession) error {
 	// At this stage the KafkaSource instance is considered Ready (TODO: update KafkaSource status)
 	return nil
 }
+
+// Default retry configuration, 5 retries, exponential backoff with 50ms delay
```

Comment:
How do these settings compare with the timeout for a normal delivery?

I'm thinking of a scenario where something is misconfigured and you get a "fast failure", like a DNS NXDOMAIN, TCP RST (port closed), or a temporary 404 from gateway misconfiguration.

With these settings, a message will get dead-lettered in less than 2s (50+100+200+400+800ms). I might try starting at at 0.5s with a 4x multiplier and 4 retries (0.5+2+8+32 = 42.5s for things to stabilize).